app/configurator/components/layout-configurator.tsx

```diff
@@ -114,31 +112,12 @@ const LayoutSharedFiltersConfigurator = () => {
   const { layout } = state;
   const { sharedFilters, potentialSharedFilters } =
     useDashboardInteractiveFilters();
-  const locale = useLocale();
-  const cubeFilters = useMemo(() => {
-    return uniqBy(
-      state.chartConfigs.flatMap((config) =>
-        config.cubes.map((x) => ({
-          iri: x.iri,
-          joinBy: x.joinBy,
-          loadValues: true,
-        }))
-      ),
-      "iri"
-    );
-  }, [state.chartConfigs]);
-  const [data] = useDataCubesComponentsQuery({
-    variables: {
-      sourceType: state.dataSource.type,
-      sourceUrl: state.dataSource.url,
-      locale: locale,
-      cubeFilters: cubeFilters,
-    },
-  });
+
+  const { data } = useAllCubesComponents(state);
```

Comment:
@ptbrowne I am not sure if I understand the issue with the old code, can you explain a bit what was wrong there? It looks like "under the hood" of the `useDataCubesComponentsQuery` hook we actually also iterate over cube variables like in the new code, so technically we should also fetch the values for each cube individually?

Is the problem that we spread the dimensions and measures then, so we might need to create another map per `cubeIri`? I wonder if this logic wouldn't be better suited to pass a param to `useDataCubesComponentsQuery`, like `"multipleCubes": boolean`?

Or maybe it's fine like now, but maybe it would make sense not introduce another data fetching wrapper, but reuse what we have already (with benefits like `keepPreviousData`, etc), let me know 😅